packages/cli/generation/remote-generation/remote-workspace-runner/src/publishDocs.ts

```diff
@@ -41,184 +44,146 @@ export async function publishDocs({
+                }
+            } else {
+                const startDocsRegisterResponse = await fdr.docs.v2.write.startDocsRegister({
+                    domain,
+                    customDomains,
+                    authConfig,
+                    apiId: "",
+                    orgId: organization,
+                    filepaths,
+                    images
+                });
+                if (startDocsRegisterResponse.ok) {
+                    docsRegistrationId = startDocsRegisterResponse.body.docsRegistrationId;
+                    await uploadFiles(
+                        startDocsRegisterResponse.body.uploadUrls,
+                        docsWorkspace.absoluteFilepath,
+                        context,
+                        UPLOAD_FILE_BATCH_SIZE
+                    );
+                    return convertToFilePathPairs(
+                        startDocsRegisterResponse.body.uploadUrls,
+                        docsWorkspace.absoluteFilepath
+                    );
+                } else {
+                    return startDocsRegisterFailed(startDocsRegisterResponse.error, context);
+                }
+            }
+        },
+        async ({ ir, snippetsConfig }) => {
+            const apiDefinition = convertIrToFdrApi({ ir, snippetsConfig });
+            context.logger.debug("Calling registerAPI... ", JSON.stringify(apiDefinition, undefined, 4));
+            const response = await fdr.api.v1.register.registerApiDefinition({
+                orgId: organization,
+                apiId: ir.apiName.originalName,
+                definition: apiDefinition
+            });
+
+            if (response.ok) {
+                context.logger.debug(`Registered API Definition ${response.body.apiDefinitionId}`);
+                return response.body.apiDefinitionId;
+            } else {
+                switch (response.error.error) {
+                    case "UnauthorizedError":
+                    case "UserNotInOrgError": {
+                        return context.failAndThrow(
+                            "You do not have permissions to register the docs. Reach out to support@buildwithfern.com"
+                        );
                     }
+                    default:
+                        return context.failAndThrow("Failed to register API", response.error);
                 }
-            })
-        );
-    }
+            }
+        }
+    );
+
+    const docsDefinition = await resolver.resolve();
 
-    // postprocess markdown files after uploading all images to replace the image paths in the markdown files with the fileIDs
-    for (const [relativePath, markdown] of Object.entries(parsedDocsConfig.pages)) {
-        parsedDocsConfig.pages[RelativeFilePath.of(relativePath)] = replaceImagePaths(markdown, collectedFileIds);
+    if (docsRegistrationId == null) {
+        return context.failAndThrow("Failed to publish docs.", "Docs registration ID is missing.");
```

Comment:
Don't think we ever want to show the error `"Docs registration ID is missing."` is a user

Reply:
will fix in a follow up